code/modules/chemistry/Chemistry-Tools.dm

```diff
@@ -817,11 +826,66 @@ proc/ui_describe_reagents(atom/A)
 				src.add_reagents_to_containers(reagent, divided_amount - remaining_container_space, sdata, temp_new, donotreact, donotupdate)  //...then run the whole proc again with the remaining reagent, evenly distributing to remaining containers
 			else
 				container.reagents.add_reagent(reagent, divided_amount, sdata, temp_new, donotreact, donotupdate)
-
 	disposing()
 		src.remove_container()
 		. = ..()
 
+	fractional
+		max_amount_of_containers = 4
+		var/container_order[4]
+		desc = "A set of glass tubes, conveniently capable of splitting the outputs of more advanced reactions. Can be hooked up to many types of containers."
+		icon_state = "condenser_fractional"
+		fluid_prefix = "condenser_fractional"
+
+		add_reagents_to_containers(reagent, amount, sdata, temp_new, donotreact, donotupdate, priority)
+			var/obj/chosen_container
+			if (priority <= max_amount_of_containers)
+				chosen_container = container_order[priority]
+
+			if(!chosen_container || chosen_container.reagents.maximum_volume <= chosen_container.reagents.total_volume)	//all full? backflow!!
+				src.reagents.add_reagent(reagent, amount, sdata, temp_new, donotreact, donotupdate)
+				return
+
+			var/remaining_container_space = chosen_container.reagents.maximum_volume - chosen_container.reagents.total_volume
+			if(remaining_container_space < amount) 																			//if there's more reagent to add than the beaker can hold...
+				chosen_container.reagents.add_reagent(reagent, remaining_container_space, sdata, temp_new, donotreact, donotupdate) //...add what we can to the beaker...
+				src.reagents.add_reagent(reagent, amount, sdata, temp_new, donotreact, donotupdate)
+			else
+				chosen_container.reagents.add_reagent(reagent, amount, sdata, temp_new, donotreact, donotupdate)
+
+		add_line(var/obj/container, var/containerNo)
+			var/x_off = 5*((containerNo+1)%2)  //alternate between 0/+5
+			var/y_off = (4*round((containerNo-1)/2)) //go up by 4 pixels for every 2nd connection
+			var/datum/lineResult/result = drawLine(src, container, "condenser_[containerNo]", "condenser_end_[containerNo]", src.pixel_x-4+x_off, src.pixel_y+5+y_off, container.pixel_x, container.pixel_y + container.get_chemical_effect_position(), mode=LINEMODE_STRETCH_NO_CLIP)
+			result.lineImage.layer = src.layer+0.01
+			result.lineImage.pixel_x = -src.pixel_x
+			result.lineImage.pixel_y = -src.pixel_y
+			src.UpdateOverlays(result.lineImage, "tube\ref[containerNo]")
+		remove_container(var/obj/container)
+			for(var/i= 1 to max_amount_of_containers)
+				if (container_order[i] == container)
+					src.connected_containers.Remove(container)
+					src.UpdateOverlays(null, "tube\ref[i]")
+					UnregisterSignal(container, COMSIG_ATTACKHAND)
+					UnregisterSignal(container, XSIG_OUTERMOST_MOVABLE_CHANGED)
+					UnregisterSignal(container, COMSIG_MOVABLE_MOVED)
+					container_order[i] = FALSE
+
+		add_container(var/obj/container)
```

Comment:
needs to either check for or handle two outputs being connected to the same target, currently just runtimes